opslevel/resource_opslevel_property_assignment.go

```diff
@@ -0,0 +1,124 @@
+package opslevel
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/opslevel/opslevel-go/v2023"
+)
+
+func resourcePropertyAssignment() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages properties assigned to entities (like Services)",
+		Create:      wrap(resourcePropertyAssignmentCreate),
+		Update:      wrap(resourcePropertyAssignmentUpdate),
+		Read:        wrap(resourcePropertyAssignmentRead),
+		Delete:      wrap(resourcePropertyAssignmentDelete),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"last_updated": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"definition": {
+				Type:        schema.TypeString,
+				Description: "The custom property definition's ID or alias.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"owner": {
+				Type:        schema.TypeString,
+				Description: "The ID or alias of the entity that the property has been assigned to.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"value": {
+				Type:        schema.TypeString,
+				Description: "The value of the custom property.",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourcePropertyAssignmentCreate(d *schema.ResourceData, client *opslevel.Client) error {
+	input := opslevel.PropertyInput{
+		Owner:      *opslevel.NewIdentifier(d.Get("owner").(string)),
+		Definition: *opslevel.NewIdentifier(d.Get("definition").(string)),
+		Value:      opslevel.JSONString(d.Get("value").(string)),
+	}
+
+	resource, err := client.PropertyAssign(input)
+	if err != nil {
+		return err
+	}
+	d.SetId(fmt.Sprintf("%s:%s", resource.Owner.Id(), resource.Definition.Id))
+
+	return resourcePropertyAssignmentRead(d, client)
+}
+
+func resourcePropertyAssignmentUpdate(d *schema.ResourceData, client *opslevel.Client) error {
+	// TODO: this is a hack
+	// cannot update an existing property assignment, so instead
+	// delete the current assignment and create a new one.
+	id := strings.Split(d.Id(), ":")
+	ownerId := id[0]
+	definitionId := id[1]
```

Comment:
This needs more validation logic or its prone to error - what if a customer does `terraform import opslevel_property xxx_id_xxx`

Check out - https://github.com/OpsLevel/terraform-provider-opslevel/blob/main/opslevel/resource_opslevel_service_tool.go#L111 - for a strong example

Reply:
https://github.com/OpsLevel/terraform-provider-opslevel/pull/174/commits/43d94314ce296fb2173e5107aadd65e6932d255e pushed new commit to do this